src/component/Tooltip.tsx

```diff
@@ -202,101 +196,120 @@ export class Tooltip<TValue extends ValueType, TName extends NameType> extends P
-          tooltipDimension: boxHeight,
-          viewBoxDimension: viewBox.height,
-        });
-      } else {
-        outerStyle.visibility = 'hidden';
-      }
+      translateY = getTranslate({
+        key: 'y',
+        tooltipDimension: boxHeight,
+        viewBoxDimension: viewBox.height,
+      });
+    } else {
+      outerStyle.visibility = 'hidden';
     }
+  }
+  outerStyle = {
+    ...translateStyle({
+      transform: useTranslate3d
+        ? `translate3d(${translateX}px, ${translateY}px, 0)`
+        : `translate(${translateX}px, ${translateY}px)`,
+    }),
+    ...outerStyle,
+  };
 
+  if (isAnimationActive && active) {
     outerStyle = {
       ...translateStyle({
-        transform: this.props.useTranslate3d
-          ? `translate3d(${translateX}px, ${translateY}px, 0)`
-          : `translate(${translateX}px, ${translateY}px)`,
+        transition: `transform ${animationDuration}ms ${animationEasing}`,
       }),
       ...outerStyle,
     };
+  }
 
-    if (isAnimationActive && active) {
-      outerStyle = {
-        ...translateStyle({
-          transition: `transform ${animationDuration}ms ${animationEasing}`,
-        }),
-        ...outerStyle,
-      };
-    }
+  const cls = classNames(CLS_PREFIX, {
+    [`${CLS_PREFIX}-right`]: isNumber(translateX) && coordinate && isNumber(coordinate.x) && translateX >= coordinate.x,
+    [`${CLS_PREFIX}-left`]: isNumber(translateX) && coordinate && isNumber(coordinate.x) && translateX < coordinate.x,
+    [`${CLS_PREFIX}-bottom`]:
+      isNumber(translateY) && coordinate && isNumber(coordinate.y) && translateY >= coordinate.y,
+    [`${CLS_PREFIX}-top`]: isNumber(translateY) && coordinate && isNumber(coordinate.y) && translateY < coordinate.y,
+  });
 
-    const cls = classNames(CLS_PREFIX, {
-      [`${CLS_PREFIX}-right`]:
-        isNumber(translateX) && coordinate && isNumber(coordinate.x) && translateX >= coordinate.x,
-      [`${CLS_PREFIX}-left`]: isNumber(translateX) && coordinate && isNumber(coordinate.x) && translateX < coordinate.x,
-      [`${CLS_PREFIX}-bottom`]:
-        isNumber(translateY) && coordinate && isNumber(coordinate.y) && translateY >= coordinate.y,
-      [`${CLS_PREFIX}-top`]: isNumber(translateY) && coordinate && isNumber(coordinate.y) && translateY < coordinate.y,
-    });
+  return (
+    // ESLint is disabled to allow listening to the `Escape` key. Refer to
+    // https://github.com/recharts/recharts/pull/2925
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+    <div
+      role="dialog"
+      onKeyDown={event => {
+        if (event.key === 'Escape') {
+          setState(prev => ({
+            ...prev,
+            dismissed: true,
+            dismissedAtCoordinate: {
+              ...prev.dismissedAtCoordinate,
+              x: coordinate.x,
+              y: coordinate.y,
+            },
+          }));
+        }
+      }}
+      className={cls}
+      style={outerStyle}
+      ref={wrapperNode}
```

Comment:
I am not sure whether this change equivalent.

Reply:
Not sure either - this is the approach I would take with refs in general (and how they should be done now), but might be a breaking change. Will test to see if it is. Can revert to a legacy ref if needed